lib-router/src/main/java/com/github/sasd97/lib_router/BaseRouter.java

```diff
@@ -1,27 +1,45 @@
+// Copyright (c) 2018 by Alexander Dadukin (st235@yandex.ru)
+// All rights reserved.
+
 package com.github.sasd97.lib_router;
 
 import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 
 import com.github.sasd97.lib_router.commands.Command;
 import com.github.sasd97.lib_router.exceptions.SatelliteNotAttachedException;
 import com.github.sasd97.lib_router.satellites.Satellite;
 
-/**
- * Created by alexander on 06/09/2017.
- */
+import java.util.ArrayList;
+import java.util.List;
 
 public class BaseRouter implements Router {
 
-    private Satellite satellite;
+    private final List<Satellite> satellites = new ArrayList<>();
 
     @Override
     public void pushCommand(@NonNull Command command) {
-        if (satellite == null) throw new SatelliteNotAttachedException();
-        satellite.execute(command);
+        Satellite applicableSatellite = findSatellite(command);
+        if (applicableSatellite == null) throw new SatelliteNotAttachedException();
```

Comment:
may be "SatelliteNotAddedException" would be better?

Reply:
I would prefer `attached`

Reply:
ok